packages/textfield/src/mwc-textfield-base.ts

```diff
@@ -486,11 +514,29 @@ export abstract class TextFieldBase extends FormElement {
       hasOutline: () => Boolean(this.outlineElement),
       notchOutline: (labelWidth) => {
         const outlineElement = this.outlineElement;
-        if (outlineElement) {
+        if (outlineElement && !this.outlineOpen) {
           this.outlineWidth = labelWidth;
           this.outlineOpen = true;
         }
       }
     };
   }
+
+  protected async onLabelChange() {
+    if (this.label) {
+      await this.layout();
+    }
+  }
+
+  async layout() {
+    await this.updateComplete;
+
+    if (this.labelElement && this.outlineElement) {
+      // sometimes notchOutline in adapter is called before label is rendered.
```

Comment:
Can you expand the explanation of why we need this method (similar to what's in the PR description would be good: "If X is initially display:none or otherwise has the wrong width, then the foundation will calculate the wrong size and do its initial layout incorrectly, etc."